File: Curso_04-Python3/calculadora.py
def soma(n1,n2):
    print(f" Soma: {n1} + {n2}")
    resultado = n1 + n2
    return resultado

def subtracao(n1, n2):
     print(f" Subtraçao: {n1} - {n2}")
     resultado = n1 - n2
     return resultado


def multiplicacao(n1, n2):
    print(f" Multiplicaçao: {n1} * {n2}")
    resultado = n1 * n2
    return resultado


def divisao(n1, n2):
    print(f" Divisao: {n1} / {n2}")
    resultado = n1 / n2
    return resultado


n1 = int(input("Digite o primeiro numero: "))
n2 = int(input("Digite o segundo numero: "))

operador = input("Digite o operador ")

if operador == '+':
    resultado = soma (n1, n2)
elif operador == '-':
      resultado = subtracao (n1, n2)
elif operador == '*':
      resultado = multiplicacao (n1, n2)
elif operador == '/':
       resultado = divisao (n1, n2)
else:
    print("Operador incorreto: ")

    print(f" Resultado é:  {resultado}" )

File: Curso_03-Kotlin/src/Calculadora.kt
import java.math.BigDecimal
fun calculaOperacao(primeiroNumero: BigDecimal, operador: String, segundoNumero: BigDecimal): BigDecimal {
    when(operador){
        "+" -> return primeiroNumero + segundoNumero
        "-" -> return primeiroNumero - segundoNumero
        "*" -> return primeiroNumero * segundoNumero
        "/" -> return primeiroNumero / segundoNumero
        "%" -> return primeiroNumero % segundoNumero
        else -> return BigDecimal.ZERO

    }
}

fun main() {
    println(" Digite um número: ")
    val primeiroNumero = readLine()?.toBigDecimalOrNull()

    if(primeiroNumero == null){
        println("Primeiro numero invalido: ")
        return
    }

    println("Digite o operador +, -, *, /, % :")
    val operador = readLine()
    if(operador.isNullOrEmpty() || operador.isNullOrBlank() ||
        (operador !="+" && operador !="-" && operador !="*" && operador !="/" && operador !="%")){
        println("Operador invalido: ")
        return
    }

    println(" Digite o Segundo número: ")
    val segundoNumero = readLine()?.toBigDecimalOrNull()

    if(segundoNumero == null){
        println(" Segundo numero invalido: ")
        return
    }

    val resultado = calculaOperacao(primeiroNumero, operador, segundoNumero)
    println("O Resultado da sua operação foi: $resultado ")
}